optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/tri/BavetFilterTriConstraintStream.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.score.stream.bavet.tri;
+
+import org.optaplanner.core.api.function.TriPredicate;
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.impl.score.stream.bavet.BavetConstraint;
+import org.optaplanner.core.impl.score.stream.bavet.common.BavetNodeBuildPolicy;
+
+public final class BavetFilterTriConstraintStream<Solution_, A, B, C> extends BavetAbstractTriConstraintStream<Solution_, A, B, C> {
+
+    private final TriPredicate<A, B, C> predicate;
+
+    public BavetFilterTriConstraintStream(BavetConstraint<Solution_> bavetConstraint, TriPredicate<A, B, C> predicate) {
+        super(bavetConstraint);
+        this.predicate = predicate;
+        if (predicate == null) {
+            throw new IllegalArgumentException("The predicate (null) cannot be null.");
+        }
```

Comment:
Order of the check and the assignment could be switched here.

Reply:
You mean "if (null == predicate)"?
The rest of optaplanner doesn't follow that convention.

Reply:
No, if I understand it correctly, it's about checking for null first and doing the assignment `this.predicate = predicate;` just after that.